### PR TITLE
Update JIT paths and build configurations

### PR DESCRIPTION
**Pull request by JoGei**
- RISCV example: Enable FPU in toolchain configuration
- GCCJIT: Add to RPATH settings to linking of JIT code with external libraries
- Windows builds: Force build of external JIT DLLs to include symbols

### CODE REVIEW

File: examples/SW/riscv/cmake/pulpino_tumeda/toolchain.cmake [hunk 1]
```diff
@@ -16,7 +16,7 @@ set(CMAKE_CXX_COMPILER ${RISCV_ELF_GCC_PREFIX}/bin/${RISCV_ELF_GCC_BASENAME}-g++
 set(CMAKE_OBJCOPY ${RISCV_ELF_GCC_PREFIX}/bin/${RISCV_ELF_GCC_BASENAME}-objcopy${EXE_EXT})
 
 add_definitions(-D__riscv__)
-add_definitions(-march=rv32imac)
-add_definitions(-mabi=ilp32)
+add_definitions(-march=rv32gc)
+add_definitions(-mabi=ilp32f)
```
Comment (rafzi) [inline examples/SW/riscv/cmake/pulpino_tumeda/toolchain.cmake:20]: any reason not to switch directly to ilp32d?

File: src/jitlibs/resources/CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,14 @@
 PROJECT(resources)
 
+CMAKE_HOST_SYSTEM_INFORMATION(RESULT HOSTINFO QUERY 
+	OS_NAME 	#0
+)
+LIST(GET HOSTINFO 0 HI_OS_NAME)
+
+IF(${HI_OS_NAME} STREQUAL "Windows")
```
Comment (rafzi) [inline src/jitlibs/resources/CMakeLists.txt:8]: the conventional way for this check is "IF(WIN32)"

File: src/jitlibs/softfloat/CMakeLists.txt [hunk 3]
```diff
@@ -29,35 +29,29 @@ LIST(GET HOSTINFO 1 HIF_SSE2)
...
-			SET(PLATFORM_N Win32-SSE2-MinGW)
-			SET(SPECIALIZE 8086-SSE)			
+			SET(PLATFORM_N Win32-SSE2-MinGW)		
 		ELSEIF(${HIF_SSE})
 			SET(PLATFORM_N Win32-MinGW)
-			SET(SPECIALIZE 8086-SSE)
 		ELSE()
-			SET(PLATFORM_N Win32-MinGW)
-			SET(SPECIALIZE 8086)			
+			SET(PLATFORM_N Win32-MinGW)
```
Comment (rafzi) [inline src/jitlibs/softfloat/CMakeLists.txt:36]: duplicate branch body

File: src/jitlibs/softfloat/CMakeLists.txt [hunk 4]
```diff
@@ -380,16 +374,19 @@ SET(SRCS_OTHERS
...
 ADD_LIBRARY(${BPROJECT_NAME} STATIC
     ${SRCS_PRIMITIVES}
     ${SRCS_SPECIALIZE}
     ${SRCS_OTHERS}
 )
-TARGET_COMPILE_OPTIONS(${BPROJECT_NAME} PRIVATE
-	-O3           
-)
+
+IF(${HI_OS_NAME} STREQUAL "Windows")
```
Comment (rafzi) [inline src/jitlibs/softfloat/CMakeLists.txt:384]: IF(WIN32)

**Review by rafzi — APPROVED** [review]
thanks, looks good to merge! some minor comments inline

File: examples/SW/riscv/cmake/pulpino_tumeda/toolchain.cmake (hunk 1 above)
Comment (JoGei) [reply examples/SW/riscv/cmake/pulpino_tumeda/toolchain.cmake]: none. fixed.

**Pull request merged**